### PR TITLE
qemu: use shared libslirp instead of vendored

### PR DESCRIPTION
**Pull request by zowoq**
###### Motivation for this change

So we don't need to patch the vendored `libslirp`.

https://github.com/NixOS/nixpkgs/pull/76065
https://github.com/NixOS/nixpkgs/pull/79050

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @risicle

### CODE REVIEW

File: pkgs/applications/virtualization/qemu/default.nix [hunk 1]
```diff
@@ -54,7 +54,7 @@ stdenv.mkDerivation rec {
   buildInputs =
     [ zlib glib perl pixman
       vde2 texinfo makeWrapper lzo snappy
-      gnutls nettle curl
+      gnutls nettle curl libslirp
```
Comment (andir) [inline pkgs/applications/virtualization/qemu/default.nix:57]: Is this a strictly required dependency of some sorts? If not please give it a feature flag that defaults to `!nixosTestRunner` like with many of the features above.

**Review by andir — CHANGES_REQUESTED** [review]
Can you add the reasoning you hinted at in the PR description also to the commit that adds the feature to QEMU? It would be nice to be able to judge why this was done just from the commit logs without yet another indirection.

File: pkgs/applications/virtualization/qemu/default.nix (hunk 1 above)
Comment (andir) [reply pkgs/applications/virtualization/qemu/default.nix:57]: Oh, I see, we've been using the vendored version previously. So this is likely okay to stay as is.

Comment (zowoq) [reply pkgs/applications/virtualization/qemu/default.nix:57]: It does seem to be possible to completly disable libslirp but as we've already been using the vendored copy I'd rather just swap them out.

Comment (andir) [reply pkgs/applications/virtualization/qemu/default.nix:57]: Yeah that is reasonable. This actually also shrinks the closure by 0.1MB ;)

**Comment by mweinelt** [comment]
> Can you add the reasoning you hinted at in the PR description also to the commit that adds the feature to QEMU? It would be nice to be able to judge why this was done just from the commit logs without yet another indirection.

The reason would likely be 

> So we don't need to patch the vendored libslirp.

Not sure why you left that out.

**Comment by zowoq** [comment]
I think `use shared libslirp instead of vendored` is accurate and sufficient as a commit message.

**Comment by andir** [comment]
> I think `use shared libslirp instead of vendored` is accurate and sufficient as a commit message.

But it doesn't contain the *why* we did it. Stating what has been done can ofte be extracted from the changes but not the reason for that change.

Let me propose the following:

```
qemu: use shared libslirp


In the past we had to patch the bundled libslurp multiple times (e.g. #76065 &
#79050) while we have & had a proper package for libslurp. With this change we
can apply patches for libslurp once in a straight forward fashion and those
will also apply to QEMU.
```

**Comment by grahamc** [comment]
I'm a bit disappointed to see you close the PR and delete the branch over what looks like a really good commit message suggestion...

**Comment by zowoq** [comment]
And I'm disappointed that a commit message that does explain the change is being bikeshedded.

Heaps of stuff gets merged with commit messages vastly worse than this.

If what I had isn't good enough I've got other things to do.

**Comment by grahamc** [comment]
Okay, thanks.

**Comment by andir** [comment]
Just as reference: I am not trying to pick on you but I am trying to make sure everything I handle follows a good style. We even have that documented in the [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#writing-good-commit-messages)

**Comment by risicle** [comment]
WFM macos 10.14, LGT ... oh.

**Comment by zowoq** [comment]
I was being serious when I said "I've got other things to do rather than argue" but I see that this is now being discussed on discourse so I will respond. I'm not going to open an account on discourse.

> I am not trying to pick on you but I am trying to make sure everything I handle follows a good style.

If you want people to follow a style, find a way for it to be enforced consistently tree-wide. Otherwise it isn't anything except unsustainable pedantry.

We even have that documented in the [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#writing-good-commit-messages)

As we don't even enforce `format the commit messages in the following way` from `CONTRIBUTING` I'm actually kind of annoyed by `writing good commit messages` being used as some sort of justification here.

> During a review that I did today a contributor decided to close their PR 

s/contributor/committer/

**Comment by andir** [comment]
Yet again I did not mean to offend you.

You made me think about my approach. Maybe I am expecting too much? That is
why I wanted to state my view on it and invite everyone to state theirs.
Discourse is, even thought I would prefer a proper ML, the discussion
platform that is currently established in our community. Thus that seemed
the the reasonable place to bring this up.

Also I am thinking about contributors as super class of everyone that
contributes to the project , including committers. This includes you and
everyone that once had or still has commit access to the repo.

**Comment by mweinelt** [comment]
Can we pick this up, please? If we had merge this we'd not be affected by CVE-2020-29129.

**Comment by zowoq** [comment]
I'll reopen if the request to change the commit message is withdrawn.

**Comment by andir** [comment]
I'll not withdraw that request. It should clearly state the motivation for this change.

**Comment by risicle** [comment]
I'm losing the will to live here.

**Pull request closed without merging**